crates/ty_python_semantic/resources/mdtest/subscript/tuple.md

```diff
@@ -69,8 +69,60 @@ def _(m: int, n: int):
     t[::0]  # error: [zero-stepsize-in-slice]
 
     tuple_slice = t[m:n]
-    # TODO: Should be `tuple[Literal[1, 'a', b"b"] | None, ...]`
-    reveal_type(tuple_slice)  # revealed: tuple[Unknown, ...]
+    reveal_type(tuple_slice)  # revealed: tuple[Literal[1, "a", b"b"] | None, ...]
+```
+
+## Slices of homogeneous and mixed tuples
+
+```toml
+[environment]
+python-version = "3.11"
+```
+
+```py
+from typing import Literal
+
+def homogeneous(t: tuple[str, ...]) -> None:
+    reveal_type(t[0])  # revealed: str
+    reveal_type(t[1])  # revealed: str
+    reveal_type(t[2])  # revealed: str
+    reveal_type(t[3])  # revealed: str
+
+    reveal_type(t[-1])  # revealed: str
+    reveal_type(t[-2])  # revealed: str
+    reveal_type(t[-3])  # revealed: str
+    reveal_type(t[-4])  # revealed: str
+
+def mixed(t: tuple[Literal[1], Literal[2], *tuple[str, ...], Literal[9], Literal[10]]) -> None:
+    reveal_type(t[0])  # revealed: Literal[1]
+    reveal_type(t[1])  # revealed: Literal[2]
+    reveal_type(t[2])  # revealed: str
+    reveal_type(t[3])  # revealed: str
```

Comment:
Similarly here, wouldn't this be better as `str | Literal[9, 10]`? If the variadic middle element is length 0, it'll be `Literal[10]`; if it's length 1, it'll be `Literal[9]`; and if it's length >=2, it'll be `str`

Reply:
ditto